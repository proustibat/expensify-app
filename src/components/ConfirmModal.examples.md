```js
class ModalExample extends React.Component {
    constructor() {
        super()
        this.state = { isModalOpen: false };
        this.handleConfirm = this.handleConfirm.bind(this);
        this.handleDecline = this.handleDecline.bind(this);
        this.openModal = this.openModal.bind(this);
        this.closeModal = this.closeModal.bind(this);
    }
    
    openModal(){
        this.setState( () => ({
            isModalOpen: true
        }))
    }
  
    closeModal(){
        this.setState( () => ({
            isModalOpen: false
        }))
    }
  
    handleConfirm() {
        alert( 'User said yes' );
        this.closeModal();
    }
  
    handleDecline() {
        alert( 'User said No' );
        this.closeModal();
    }
  
    render() {
        return (
            <div>
                <button onClick={this.openModal}>Open</button>
                <ConfirmModal
                    showModal = { this.state.isModalOpen }
                    handleCloseModal = { this.handleDecline }
                    handleConfirm = { this.handleConfirm }
                    body="Are you sure?"
                    confirmText="Yes"
                    cancelText="No"
                />
            </div>
        );
    }
};
<ModalExample />
```

