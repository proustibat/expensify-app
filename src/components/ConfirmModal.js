import React from 'react';
import Modal from 'react-modal';

export default class ConfirmModal extends React.Component {

    customStyles = {
        content : {
            top                   : '50%',
            left                  : '50%',
            right                 : 'auto',
            bottom                : 'auto',
            marginRight           : '-50%',
            transform             : 'translate(-50%, -50%)'
        }
    };

    constructor( props ) {
        super( props );

        // Make sure to bind modal to your appElement (http://reactcommunity.org/react-modal/accessibility/)
        Modal.setAppElement( document.querySelector( '#app' ) );
    }

    render() {
        return (
            <Modal
                isOpen = { this.props.showModal }
                contentLabel = { 'Confirm choice modal' }
                style = { this.customStyles}
                onRequestClose = { this.props.handleCloseModal }
            >
                {/*Are you sure you wanna delete this expense?*/}
                <p>{ this.props.body }</p>
                <div className="button-container--space-around">
                    <button className="button button--cancel" onClick={ this.props.handleCloseModal }>{ this.props.cancelText }</button>
                    <button className="button button--valid" onClick={ this.props.handleConfirm }>{ this.props.confirmText }</button>
                </div>
            </Modal>
        );
    }
}

