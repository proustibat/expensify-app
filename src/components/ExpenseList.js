import React from 'react';
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import getVisibleExpenses from '../selectors/expenses';

export const ExpenseList = props => (
    <div className="content-container">
        <div className="list-header">
            <div className="show-only-on-mobile">Expenses</div>
            <div className="show-only-on-desktop">Expense</div>
            <div className="show-only-on-desktop">Amount</div>
        </div>

        <div className="list-body">
            { props.expenses.length === 0 ? (
                <div className=" list-item list-item--message">
                    <span>No expenses</span>
                </div>
            ) : (
                props.expenses.map( expense => (
                    <ExpenseListItem
                        key = { expense.id }
                        { ...expense }
                    />
                ) )
            ) }
        </div>
    </div>
);

const mapStateToProps = state => ( {
    expenses: getVisibleExpenses( state.expenses, state.filters )
} );

const ConnectedExpenseList = connect( mapStateToProps )( ExpenseList );

export default ConnectedExpenseList;
