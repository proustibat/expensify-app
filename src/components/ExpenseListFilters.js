import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate } from "../actions/filters";
import { DateRangePicker } from 'react-dates';

export class ExpenseListFilters extends React.Component {
    state = {
        calendarFocused: null
    };

    onDatesChange = ( { startDate, endDate } ) => {
        this.props.setStartDate( startDate );
        this.props.setEndDate( endDate );
    };

    onFocusChange = calendarFocused => {
        this.setState( () => ( { calendarFocused } ) );
    };

    onTextChange = e => {
        this.props.setTextFilter( e.target.value );
    };

    onSortChange = e => {
        e.target.value === 'date' ? this.props.sortByDate() : this.props.sortByAmount();
    };

    render() {
        return(
            <div className="content-container">
                <div className="input-group">
                    <div className="input-group__item">
                        <input
                            placeholder="Search expenses"
                            type="text"
                            className="text-input"
                            value = { this.props.filters.text }
                            onChange = { this.onTextChange }
                        />
                    </div>
                    <div className="input-group__item">
                        <select
                            className="select"
                            value = { this.props.filters.sortBy }
                            onChange = { this.onSortChange }
                        >
                            <option value='date'>Date</option>
                            <option value='amount'>Amount</option>
                        </select>
                    </div>
                    <div className="input-group__item">
                        <DateRangePicker
                            startDate = { this.props.filters.startDate } // momentPropTypes.momentObj or null,
                            endDate = { this.props.filters.endDate } // momentPropTypes.momentObj or null,
                            onDatesChange = { this.onDatesChange } // PropTypes.func.isRequired,
                            focusedInput = { this.state.calendarFocused } // PropTypes.oneOf([START_DATE, END_DATE]) or null,
                            onFocusChange = { this.onFocusChange } // PropTypes.func.isRequired,
                            showClearDates = { true }
                            numberOfMonths = { 1 }
                            isOutsideRange = { () => false }
                        />
                    </div>
                </div>
            </div>
        )
    }

}

const mapStateToProps = state => ( {
    filters: state.filters
} );

const mapDispatchToProps = dispatch => ( {
    setTextFilter: text => dispatch( setTextFilter( text ) ),
    sortByDate: () => dispatch( sortByDate() ),
    sortByAmount: () => dispatch( sortByAmount() ),
    setStartDate: startDate => dispatch( setStartDate( startDate ) ),
    setEndDate: endDate => dispatch( setEndDate( endDate ) )
} );

export default connect( mapStateToProps, mapDispatchToProps )( ExpenseListFilters );
