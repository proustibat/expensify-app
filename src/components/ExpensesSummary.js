import React from "react";
import { connect } from "react-redux";
import { Link } from 'react-router-dom';
import selectExpensesTotal from "../selectors/expenses-total";
import getVisibleExpenses from '../selectors/expenses';
import numeral from "numeral";


export const ExpensesSummary = ( { count, total } ) => {
    return (
        <div className="page-header">
            <div className="content-container">
                <h1 className="page-header__title">Viewing <span>{ count }</span> expense{ count > 1 ? 's' : ''} totalling <span>{ numeral( total / 100 ).format( '$0,0.00' ) }</span></h1>
                <div className="page-header__actions">
                    <Link className="button" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = state => {
    const visibleExpenses = getVisibleExpenses( state.expenses, state.filters );
    return {
        count: visibleExpenses.length,
        total: selectExpensesTotal( visibleExpenses )
    }
};

const ConnectedExpensesSummary = connect( mapStateToProps )( ExpensesSummary );

export default ConnectedExpensesSummary;
