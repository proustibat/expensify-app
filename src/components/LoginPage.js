import React from "react";
import { connect } from 'react-redux';
import { startLogin } from "../actions/auth";

export const LoginPage = props => (
    <div className="box-layout">
        <div className="box-layout__box">
            <h1 className="box-layout__title">Expensify</h1>
            <p>It's time to get your expenses under control!</p>
            <button className="button" onClick={ props.startLogin }>Login with Google</button>
        </div>
    </div>
);

const mapStateToProps = undefined;

const mapDispatchToProps = dispatch => ( {
    startLogin: () => dispatch( startLogin() )
} );

export default connect( mapStateToProps, mapDispatchToProps )( LoginPage );


