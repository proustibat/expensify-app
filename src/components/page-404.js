import React from 'react';
import { Link } from 'react-router-dom';

const Page404 = () => (
    <div>
        404 - <Link to="/">Go Home</Link>
    </div>
);

export default Page404;
