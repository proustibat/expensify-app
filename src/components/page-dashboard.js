import React from 'react';
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from "./ExpenseListFilters";
import ExpensesSummary from "./ExpensesSummary";

const PageDashboard = () => (
    <div>
        <ExpensesSummary />
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default PageDashboard
