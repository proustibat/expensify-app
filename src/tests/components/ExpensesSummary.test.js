import React from 'React';
import { shallow } from 'enzyme';
import { ExpensesSummary } from "../../components/ExpensesSummary";

test( 'Should render correctly with only one expenses', () => {
    const wrapper = shallow( <ExpensesSummary count = { 1 } total = { 999 } /> );
    expect( wrapper ).toMatchSnapshot();
} );

test( 'Should render correctly with several expenses', () => {
    const wrapper = shallow( <ExpensesSummary count = { 3 } total = { 999 } /> );
    expect( wrapper ).toMatchSnapshot();
} );
