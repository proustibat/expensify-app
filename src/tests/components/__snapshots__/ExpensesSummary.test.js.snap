// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly with only one expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       expense
       totalling 
      <span>
        $9.99
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`Should render correctly with several expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        3
      </span>
       expense
      s
       totalling 
      <span>
        $9.99
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
